src/python/turicreate/toolkits/one_shot_object_detector/util/_augmentation.py

```diff
@@ -39,6 +40,8 @@ def preview_synthetic_training_data(data,
     out : SFrame
         An SFrame of sythetically generated annotated training data.
     """
+    if (isinstance(backgrounds, _tc.SArray)  and len(backgrounds) == 0):
```

Comment:
We should probably also throw an exception if:
`background is not None and not isinstance(backgrounds, _tc.SArray)`

Then add a unit test for:
`tc.one_shot_object_detector.create(self.train, target=self.target, backgrounds='foobar')`

Reply:
Actually, this user facing function isn't doing any argument checking. That's not good. Please add tests for `data` and `target` too.

Reply:
@TobyRoseman if you look at the file _error_handling.py. It does argument checking for data as well as target.
I will add the exception when background is not None or SArrary. Also test that.

Reply:
@shreyajain17 - you're right. I missed that one line. Sorry for the confusion.